### PR TITLE
ddl: support create and drop placement policy

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: part of https://github.com/pingcap/tidb/issues/26580

### What is changed and how it works?

Proposal: [placement policy](https://github.com/pingcap/tidb/pull/26221/files) 

What's Changed:
Support create placement policy and drop placement policy basically.

What's Remained:
1: create policy and drop policy statement privilege check
2: create policy should have the constraint yaml format check
3: create policy should have the constraint compatibility check
4: drop policy should have the clean the policy dependency in it's usage.
5: drop policy should notify the pd about the db/table/partition's schedule change.

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

### Release note 

```release-note
ddl: support create and drop placement policy.
```

### CODE REVIEW

File: ddl/placement_policy_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	tk.MustGetErrCode("create placement policy X "+
+		"PRIMARY_REGION=\"cn-east-1\" "+
+		"REGIONS=\"cn-east-1,cn-east-2\" ", mysql.ErrPlacementPolicyExists)
+
+	tk.MustGetErrCode("create placement policy `X` "+
+		"PRIMARY_REGION=\"cn-east-1\" "+
+		"REGIONS=\"cn-east-1,cn-east-2\" ", mysql.ErrPlacementPolicyExists)
+
+	tk.MustExec("create placement policy if not exists X " +
```
Comment (lcwangchao) [inline ddl/placement_policy_test.go:97]: Check warnings

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -6146,3 +6147,91 @@ func (d *ddl) AlterTablePartitionAttributes(ctx sessionctx.Context, ident ast.Id
...
+		case ast.PlacementOptionFollowerCount:
+			policyInfo.Followers = opt.UintValue
+		case ast.PlacementOptionVoterCount:
+			policyInfo.Voters = opt.UintValue
+		case ast.PlacementOptionLearnerCount:
+			policyInfo.Learners = opt.UintValue
+		case ast.PlacementOptionSchedule:
+			policyInfo.Schedule = opt.StrValue
+		case ast.PlacementOptionConstraints:
+			policyInfo.Schedule = opt.StrValue
```
Comment (mjonss) [inline ddl/ddl_api.go:6169]:
```suggestion
			policyInfo.Constraints = opt.StrValue
```

**Review by morgo — APPROVED** [review]

**Pull request merged**